mockito/matchers.py

```diff
@@ -246,18 +246,22 @@ def __repr__(self):
 class ArgumentCaptor(Matcher):
     def __init__(self, matcher=None):
         self.matcher = matcher or Any()
-        self.value = None
+        self.values = list()
 
     def matches(self, arg):
         result = self.matcher.matches(arg)
         if not result:
             return
-        self.value = arg
+        self.values.append(arg)
         return True
 
+    @property
+    def value(self):
+        return self.values[-1]
```

Comment:
Returns the latest captured value. Throws `IndexError` if no argument was captured. Most probably change it to throw a custom exception if no arguments were captured. Need opinions

Reply:
From my understanding this *is* a `RuntimeError` in python as the object is in the wrong state to answer `obj.value`.  (An `AttributeError` is misleading to raise inside a `@property`.)  Either raise `RuntimeError` directly or a user exception derived from it.